modules/health_quest/app/services/health_quest/patient_generated_data/patient/resource.rb

```diff
@@ -66,59 +54,34 @@ def prepare
         end
 
         ##
-        # Builds and sets attributes on the FHIR::DSTU2::Identifier object.
-        #
-        # @return [FHIR::DSTU2::Identifier]
-        #
-        def set_identifiers
-          identifier.tap do |i|
-            i.value = user.icn
-            i.system = SYSTEM_ID
-            i.type = identifier_type
-          end
-        end
-
-        ##
-        # Build the hash for the Patient objects identifier type.
+        # Build the name array for the Patient object.
         #
-        # @return [Hash]
+        # @return [Array]
         #
-        def identifier_type
-          {
-            coding: [{
-              system: CODING_SYSTEM,
-              code: IDENTIFIER_CODE,
-              userSelected: false
-            }]
-          }
+        def name
+          [{
+            use: NAME_USE,
+            family: [user.last_name],
+            given: [user.first_name]
+          }]
         end
 
         ##
-        # Builds and sets the tag array attribute on the FHIR::DSTU2::Meta object.
+        # Return the patients ICN.
         #
-        # @return [FHIR::DSTU2::Meta]
+        # @return [String]
         #
-        def set_meta
-          meta.tap do |m|
-            m.tag = [{
-              system: META_SYSTEM,
-              code: META_CODE,
-              display: META_DISPLAY
-            }]
-          end
+        def identifier_value
+          user.icn
         end
 
         ##
-        # Build the name array for the Patient object.
+        # Return the patients identifier attribute name.
         #
-        # @return [Array]
+        # @return [String]
         #
-        def name
-          [{
-            use: NAME_USE,
-            family: [user.last_name],
-            given: [user.first_name]
-          }]
+        def identifier_code
+          'ICN'
```

Comment:
Will this always be ICN?

Reply:
@LindseySaari on a Patient resource I believe that the identifier is unique. However, I'm not sure going forward into the future and on version 4 of the FHIR spec, if that would still be the same.